atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -389,36 +436,57 @@ def do_worker_build(self, release, cluster_info, koji_upload_dir, task_id):
                 except OsbsException:
                     pass
 
+    def select_and_start_cluster(self, platform):
+        ''' Choose a cluster and start a build on it '''
+
+        config = get_config(self.workflow)
+        clusters = config.get_enabled_clusters_for_platform(platform)
+
+        if not clusters:
+            raise UnknownPlatformException('No clusters found for platform {}!'
+                                           .format(platform))
+
+        possible_clusters = None
+        retry_info = ClusterRetryContexts(clusters, self.max_cluster_fails,
+                                          self.find_cluster_retry_delay)
+
+        while True:
+            if not clusters:
+                raise RuntimeError('Could not find appropriate cluster for worker build.')
+            possible_clusters = self.get_clusters(clusters, platform, retry_info)
```

Comment:
Maybe a comment here to explain that self.get_clusters() may actually modify `clusters` here?